spec/unit/util/pidlock_spec.rb

```diff
@@ -9,6 +9,39 @@
   before(:each) do
     @lockfile = tmpfile("lock")
     @lock = Puppet::Util::Pidlock.new(@lockfile)
+    @ps_argument_for_current_kernel = @lock.send(:ps_argument_for_current_kernel)
+  end
+
+  describe "#ps pid argument on posix" do
+    it "should be -eq when kernel is Linux" do
+      allow(Facter).to receive(:value).with(:kernel).and_return("Linux")
+      expect(@lock.send(:ps_argument_for_current_kernel)).to eq("-eq")
+    end
+
+    it "should be -T when kernel is AIX" do
+      allow(Facter).to receive(:value).with(:kernel).and_return("AIX")
+      expect(@lock.send(:ps_argument_for_current_kernel)).to eq("-T")
+    end
+
+    it "should be -p when kernel is Darwin or other" do
+      allow(Facter).to receive(:value).with(:kernel).and_return("Darwin")
+      expect(@lock.send(:ps_argument_for_current_kernel)).to eq("-p")
+    end
+
+    it "should fallback to -p when current posix argument fails" do
+      if Puppet.features.posix?
```

Comment:
We usually use rspec include/exclude filters, and have been moving to using `Puppet::Util::Platform` instead of `Puppet.features` for platform detection. So something like `it "...", unless: Puppet::Util::Platform.windows? do`